src/convertToObject.js

```diff
@@ -6,7 +6,31 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  // if the sourceString is empty return an empty object
+  if (!sourceString) {
+    return {};
+  }
+
+  // remove all tab and new line characters
+  // replaceAll doesn't work on this version of Node
+  const clearedString = sourceString.replace(/[\t\n]/g, '');
+  const stylesArr = clearedString.split(';');
+  const stylesObj = {};
+
+  stylesArr.forEach((style) => {
+    let [name, value] = style.split(':');
+
+    // name and value can be empty or undefined
+    if (name && value) {
+      name = name.trim();
+      value = value.trim();
+      // add a new line character where it is needed
+      value = value.replace(/,/g, ',\n');
+      stylesObj[name] = value;
+    }
+  });
```

Comment:
[CODE PRACTICE] - Instead of using a `forEach` loop to populate `stylesObj`, consider using `reduce` method which is more suited for transforming an array into an object. This aligns with the checklist item to avoid using loops and to use iteration methods instead.